core/src/main/java/aima/core/search/basic/support/BasicBidirectionalActions.java

```diff
@@ -0,0 +1,86 @@
+package aima.core.search.basic.support;
+
+import aima.core.search.api.BidirectionalActions;
+import aima.core.search.api.Node;
+import aima.core.search.api.SearchController;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * @author manthan.
+ */
+public class BasicBidirectionalActions<A, S> implements BidirectionalActions<A> {
+    private List<A> fromInitialStatePartList = new ArrayList<>();
+    private List<A> fromGoalStatePartList = new ArrayList<>();
+    private List<A> fromInitialStateToGoalStateList = new ArrayList<>();
+    private List<A> fromGoalStateToInitialStateList = new ArrayList<>();
+    private Node<A, S> previousMeetingOfTwoFrontiers;
+    private Node<A, S> nextNodeToBeEvaluated;
+    private Node<A, S> meetingOfTwoFrontiers;
+    private boolean fromFront;
+    protected SearchController<A, S> searchController = new BasicSearchController<>();
+
```

Comment:
This class is too complex for a basic support class as well as being too coupled to the current BidirectionalSearch's implementation approach. The constructor for this class should only need to take the fromInitialListPartList and the fromGoalStatePartList and to derive the other two lists from them by joining/sorting appropriately. The other logic should migrate back into BidirectionalSearch as supporting routines.